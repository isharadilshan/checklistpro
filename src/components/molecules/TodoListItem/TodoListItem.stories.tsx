import React from 'react';
import {NativeBaseProvider} from 'native-base';
import {ComponentMeta, ComponentStory} from '@storybook/react';

import TodoListItem from './';

export default {
  title: 'components/molecules/TodoListItem',
  component: TodoListItem,
} as ComponentMeta<typeof TodoListItem>;

export const Basic: ComponentStory<typeof TodoListItem> = () => {
  const args = {
    id: 'TODO_ID',
    title: 'Toto Title',
    description: 'Todo Description',
    status: 'INPROGRESS',
    category: 'MEDICAL',
    createdDate: 1680101338199,
    updatedDate: 1680101338199,
    hideButtons: false,
    onPressEdit: () => {},
    onPressDelete: () => {},
  };
  return (
    <NativeBaseProvider>
      <TodoListItem {...args} />
    </NativeBaseProvider>
  );
};
