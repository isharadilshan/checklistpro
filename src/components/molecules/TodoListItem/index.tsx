import React from 'react';
import {Box, Badge, HStack, View, Text} from 'native-base';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import {getDateName} from '../../../utils/helper/Date';

type TodoListItemProps = {
  id: string;
  title: string;
  description: string;
  category: string;
  createdDate: number;
  onPressEdit: (id: string) => void;
  onPressDelete: (id: string) => void;
};

const TodoListItem: React.FC<TodoListItemProps> = ({
  id,
  title,
  description,
  category,
  createdDate,
  onPressEdit,
  onPressDelete,
}) => {
  return (
    <Box
      height={150}
      rounded="sm"
      overflow="hidden"
      borderColor="coolGray.200"
      borderWidth="1"
      _dark={{
        borderColor: 'coolGray.600',
        backgroundColor: '#1f2937',
      }}
      _light={{
        backgroundColor: '#1f2937',
      }}
      p={2}
      my={2}
    >
      <HStack height={'100%'}>
        <Box w={'75%'}>
          <View style={{flex: 1, justifyContent: 'space-between'}}>
            <View>
              <Text color={'blueGray.400'} fontSize="xl">
                {title}
              </Text>
              <Text color={'blueGray.400'}>{description}</Text>
              <Text mt={3} color={'blueGray.400'}>
                {getDateName(createdDate)}
              </Text>
            </View>
            <Badge width={150} variant={'outline'} colorScheme="success">
              {category}
            </Badge>
          </View>
        </Box>
        <Box w={'25%'}>
          <View
            style={{
              flex: 1,
              alignItems: 'flex-end',
              justifyContent: 'space-between',
            }}
          >
            <Icon
              name="square-edit-outline"
              size={30}
              color="#94a3b8"
              onPress={() => onPressEdit(id)}
            />
            <Icon
              name="delete"
              size={30}
              color="#94a3b8"
              onPress={() => onPressDelete(id)}
            />
          </View>
        </Box>
      </HStack>
    </Box>
  );
};

export default TodoListItem;
