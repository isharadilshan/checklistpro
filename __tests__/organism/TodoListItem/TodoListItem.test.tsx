import React from 'react';
import {describe, expect} from '@jest/globals';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import TodoListItem from '../../../src/components/organism/TodoListItem';

configure({adapter: new Adapter()});

const props = {
  item: {title: 'TEST TITLE', description: 'TEST DESCRIPTION'},
  onEdit: jest.fn(),
  onDelete: jest.fn(),
};

describe('when rendering', () => {
  const wrapper = shallow(<TodoListItem {...props} />);

  it('should render a component matching the snapshot', () => {
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
    expect(wrapper).toHaveLength(1);
  });
});
