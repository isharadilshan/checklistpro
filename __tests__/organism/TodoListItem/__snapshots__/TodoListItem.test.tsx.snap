// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when rendering should render a component matching the snapshot 1`] = `
<withTheme(Surface)
  style={
    {
      "backgroundColor": "#BBC8D6",
      "borderRadius": 4,
      "elevation": 4,
      "flexDirection": "row",
      "marginHorizontal": 16,
      "marginVertical": 8,
      "padding": 8,
    }
  }
>
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <Title>
      TEST TITLE
    </Title>
    <Subheading>
      TEST DESCRIPTION
    </Subheading>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingLeft": 10,
      }
    }
  >
    <withTheme(IconButton)
      icon="square-edit-outline"
      onPress={[Function]}
      size={25}
    />
    <withTheme(IconButton)
      icon="trash-can"
      onPress={[Function]}
      size={25}
    />
  </View>
</withTheme(Surface)>
`;
